r/src/array.cpp

```diff
@@ -323,6 +357,37 @@ inline SEXP StringArray_to_Vector(const std::shared_ptr<arrow::Array>& array) {
   return res;
 }
 
+SEXP DictionaryArray_to_Vector(arrow::DictionaryArray* dict_array) {
+  auto dict = dict_array->dictionary();
+  auto indices = dict_array->indices();
+
+  if (dict->type_id() != Type::STRING || indices->type_id() != Type::INT32) {
+    stop("Cannot convert Dictionary Array of type `%s`", dict_array->type()->ToString());
+  }
+
+  IntegerVector f = simple_Array_to_Vector<INTSXP>(indices);
+  f = f + 1;
```

Comment:
How does this compare with an in-place modification?

Reply:
It’s one more scan, i’ll make a custom function.

Reply:
✅